tests/run-make/print-target-list/Makefile

```diff
@@ -1,8 +0,0 @@
-include ../tools.mk
-
-# Checks that all the targets returned by `rustc --print target-list` are valid
-# target specifications
-all:
-	for target in $(shell $(BARE_RUSTC) --print target-list); do \
-		$(BARE_RUSTC) --target $$target --print sysroot; \
-	done
```

Comment:
Problem: I found out why we're struggling with this test. It's because this Makefile isn't actually checking what it says it's checking...

```make
# target specifications
all:
	for target in $(shell $(BARE_RUSTC) --print target-list); do \
		$(BARE_RUSTC) --target $$target --print sysroot || exit 1; \
	done
```

if we force an exit, we'll see the makefile actually fails in CI with the exact same error (https://github.com/rust-lang/rust/pull/128073#issuecomment-2266381093).